dask/distributed/tests/test_ipython_utils.py

```diff
@@ -0,0 +1,68 @@
+import time
+from subprocess import Popen, PIPE
+
+import pytest
+
+np = pytest.importorskip('numpy')
+pytest.importorskip('IPython.parallel')
+
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+
+from IPython.parallel import Client
+
+from dask.distributed import dask_client_from_ipclient
+import dask.array as da
+
+
+def setup_module():
+    start_cmd = ['ipcluster', 'start', '-n', '4']
+    cp = Popen(start_cmd, stdout=PIPE, stderr=PIPE)
+    tic = time.time()
+    while True:
+        line = cp.stderr.readline().decode()
+        if 'Engines appear to have started' in line:
+            break
+        elif 'CRITICAL | Cluster is already running' in line:
+            raise RuntimeWarning("Cluster is already running")
+        elif time.time() - tic > 50:
+            raise RuntimeError("Timeout waiting for cluster to start.")
+        time.sleep(0.1)
+
+
+def teardown_module():
+    kill_cmd = ['ipcluster', 'stop']
+    cp = Popen(kill_cmd, stdout=PIPE, stderr=PIPE)
+    tic = time.time()
+    while True:
+        line = cp.stderr.readline().decode()
+        if "Stopping cluster" in line:
+            break
+        elif time.time()-tic > 15:
+            # probably stopped or something went wrong
+            break
+        time.sleep(0.1)
```

Comment:
Okay so if I understand what IPython does correctly:
They use one  long runnig "ipcluster", and then `ClusterTestCase` cleans the engine's namespaces for each test.

We should probably have our own `ClusterTestCase` that uses something like `dask.distributed.Client.clear_workers` to clean up and inherits from `BaseZMQTestCase`. So we can be IPython agnostic.

Also running their tests requires we use `iptest`, which is nose with their nose plugin. This would be annoying for us to do.